### ISSUE TITLE
Value.Check works in an undesirable way with inheritance

### ISSUE THREAD

**Issue by a-k-g** [immutables/immutables#361]
```
import java.util.Arrays;
import java.util.Collections;
import java.util.List;

import org.immutables.value.Value;
import org.junit.Rule;
import org.junit.Test;
import org.junit.rules.ExpectedException;

public class ValueCheckInheritance {
    public abstract static class Zoo {
        public abstract List<String> animals();

        @Value.Check
        protected void checkAnimalsNonEmpty() {
            if (animals().isEmpty()) {
                throw new IllegalStateException("An empty zoo is just a park!");
            }
        }
    }

    @Value.Immutable
    public abstract static class BigZoo extends Zoo {
        public abstract List<String> cafeterias();

        @Value.Check
        protected void checkCafeteriasNonEmpty() {
            if (cafeterias().isEmpty()) {
                throw new IllegalStateException("Zero cafeterias make it a regular zoo!");
            }
        }
    }

    @Rule
    public ExpectedException exception = ExpectedException.none();

    @Test
    public void big_zoo_without_animals_should_fail_to_build() {
        exception.expect(IllegalStateException.class);
        exception.expectMessage("An empty zoo is just a park!");

        ImmutableBigZoo.builder()
                       .animals(Collections.emptyList())
                       .cafeterias(Collections.emptyList())
                       .build();
    }

    @Test
    public void big_zoo_without_cafeterias_should_fail_to_build() {
        exception.expect(IllegalStateException.class);
        exception.expectMessage("Zero cafeterias make it a regular zoo!");

        ImmutableBigZoo.builder()
                       .animals(Arrays.asList("Cheetah"))
                       .cafeterias(Collections.emptyList())
                       .build();
    }
}
```

The first test succeds, the second one fails. It's because in the generated ImmutableBigZoo, only the `@Value.Check` of the parent class is called -
```
private static ImmutableBigZoo validate(ImmutableBigZoo instance) {
    instance.checkAnimalsNonEmpty();
    return instance;
}
```

The ideal behavior would be for both the checks to be called, I think.

**Comment by elucash** [comment]
Make sense to invoke them all if they have different names and not override each other. If some ordering needed, it is possible to resort to have the same method name and use `super` invokation. Thank you for spotting this!

**Comment by elucash** [comment]
released in 2.2.7 thank you!
